modules/govuk_mtail/manifests/init.pp

```diff
@@ -0,0 +1,76 @@
+# == Class: govuk_mtail
+#
+# This class installs and sets-up Mtail
+#
+# === Parameters
+#
+# [*manage_repo_class*]
+#   Whether to use a separate repository to install Statsd
+#   Default: false (use 'govuk_ppa' repository)
+#
+# [*logs*]
+#   List of files to monitor
+#
+# [*enabled*]
+#   Enable Mtail service. (default true)
+#
+# [*port*]
+#   HTTP port to listen on. (default "3903")
+#
+# [*collectd_socketpath*]
+#   Path to collectd unixsock to write metrics to.
+#
+# [*graphite_host_port*]
+#   Host:port to graphite carbon server to write metrics to.
+#
+# [*statsd_hostport*]
+#   Host:port to statsd server to write metrics to.
+#
+# [*metric_push_interval*]
+#   Interval between metric pushes, in seconds (default 60)
+#
+# [*extra_args*]
+#   Mtail program extra arguments (default -log_dir /var/log/mtail)
+#
+class govuk_mtail(
+  $logs = [],
+  $manage_repo_class = false,
+  $enabled = false,
+  $port = 3093,
+  $collectd_socketpath = undef,
+  $graphite_host_port = undef,
+  $statsd_hostport = undef,
+  $metric_push_interval = 60,
+  $extra_args = '-log_dir /var/log/mtail',
+) {
+
+  validate_bool($manage_repo_class)
+
+  if $manage_repo_class {
+    $repo_class = 'govuk_mtail::repo'
+  } else {
+    $repo_class = 'govuk_ppa'
+  }
+
+  include $repo_class
+
+  $logs_ = hiera_array('govuk_mtail::logs', $logs)
```

Comment:
Why do we do this, rather than just passing in the `$logs` parameter?

Reply:
To make sure we do a recursive merge